### PR TITLE
Enable debug log mode & use concurrency

### PR DESCRIPTION
**Pull request by nathan-beneke**
With the `--debug` flag a user can set the log level to default.

Introduce the https://github.com/rs/zerolog logging framework for structured and leveled logging

Add @nihalgonsalves as a CODEOWNER

Restructure packages to a more typical setup. As a side-effect: combine the `gen` package with `cmd` to avoid cyclical imports, and use the `Run()` interface provided by Kong so no switch statement is necessary

Use Apache 2.0 license correctly on all go files

Speed up total runtime by fetching roles/groups/policies in parallel

### CODE REVIEW

File: pkg/util/ini.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import "gopkg.in/ini.v1"
+
+func GetKeySetter(section *ini.Section) func(key, value string) {
+	return func(key, value string) {
+		_, err := section.NewKey(key, value)
+		if err != nil {
+			panic(err)
+		}
+	}
+}
```
Comment (nihalgonsalves) [inline pkg/util/ini.go:25]:
```suggestion
import (
	"github.com/rs/zerolog/log"
	"gopkg.in/ini.v1"
)

func GetKeySetter(section *ini.Section) func(key, value string) {
	return func(key, value string) {
		_, err := section.NewKey(key, value)
		if err != nil {
			log.Panic().Err(err).
				Str("section", section.Name()).Str("key", key).Str("val", value).
				Msg("could not set")
		}
	}
}
```

**Review by nihalgonsalves — APPROVED** [review]
Other than the comments, you're missing three panics in the IAM result error handling in `getRoleArnsForGroup` and `getRoleArnsForAttachedPolicy`, looks good otherwise :rocket:

**Comment by nathan-beneke** [comment]
> Other than the comments, you're missing three panics in the IAM result error handling in `getRoleArnsForGroup` and `getRoleArnsForAttachedPolicy`, looks good otherwise 🚀

Fixed, thanks

File: pkg/util/aws.go [hunk 2]
```diff
@@ -46,17 +61,20 @@ func (ctx *AWSContext) GetRolesAndAccounts() (roleArns []string, accountMap map[
 
 	gcio, err := ctx.sts.GetCallerIdentity(&sts.GetCallerIdentityInput{})
 	if err != nil {
-		panic(err)
+		log.Panic().Err(err).Msg("coulld not get caller identity")
```
Comment (nihalgonsalves) [inline pkg/util/aws.go:64]:
```suggestion
		log.Panic().Err(err).Msg("could not get caller identity")
```

**Review by snowiow — APPROVED** [review]

**Pull request merged**